api_app/connectors_manager/serializers.py

```diff
@@ -25,6 +68,11 @@ class ConnectorConfigSerializer(serializers.Serializer):
     secrets = serializers.JSONField()
     verification = serializers.SerializerMethodField()
 
+    def validate_secrets(self, secrets):
+        serializer = SecretSerializer(data=list(secrets.items()), many=True)
+        if serializer.is_valid(raise_exception=True):
```

Comment:
the `if` is not necessary,

Reply:
what if you pass the data to the serializer like: `data=["{key_name": key, **secret_dict} for key, secret_dict in secrets.items()]`. This way you are passing to it a `dict` only and not a `tuple`.